internal/qmp-gen/parse_test.go

```diff
@@ -79,3 +174,33 @@ Got:
 		})
 	}
 }
+
+func testGenerate(t *testing.T, in []byte) []byte {
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write(in)
+	}))
+	defer s.Close()
+
+	defs, err := readDefinitions(s.URL)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	symbols, err := parse(defs)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	need := neededTypes(symbols)
```

Comment:
Specific subsets probably makes the most sense for this test.  I'll add an additional test that verifies subsets, and leave this one to verify the "neededTypes" logic.